src/packs/nzism-v36-1022-20.ts

```diff
@@ -0,0 +1,845 @@
+} from '../rules/cloudfront';
+import {
+  CloudTrailCloudWatchLogsEnabled,
+  CloudTrailEncryptionEnabled,
+  CloudTrailLogFileValidationEnabled,
+} from '../rules/cloudtrail';
+import { CloudWatchLogGroupRetentionPeriod } from '../rules/cloudwatch';
+import { DMSReplicationNotPublic } from '../rules/dms';
+import {
+  DynamoDBAutoScalingEnabled,
+  DynamoDBInBackupPlan,
+  DynamoDBPITREnabled,
+} from '../rules/dynamodb';
+import {
+  EC2EBSInBackupPlan,
+  EC2InstanceNoPublicIp,
+  EC2InstancesInVPC,
+  EC2RestrictedSSH,
+  EC2EBSVolumeEncrypted,
+} from '../rules/ec2';
+import { ECSTaskDefinitionUserForHostMode } from '../rules/ecs';
+import { EFSEncrypted, EFSInBackupPlan } from '../rules/efs';
+import { ElastiCacheRedisClusterAutomaticBackup } from '../rules/elasticache';
+import { ElasticBeanstalkManagedUpdatesEnabled } from '../rules/elasticbeanstalk';
+import {
+  ALBHttpToHttpsRedirection,
+  ALBWAFEnabled,
+  ELBCrossZoneLoadBalancingEnabled,
+  ELBLoggingEnabled,
+  ELBTlsHttpsListenersOnly,
+} from '../rules/elb';
+import { IAMPolicyNoStatementsWithAdminAccess } from '../rules/iam';
+import { KMSBackingKeyRotationEnabled } from '../rules/kms';
+import { LambdaInsideVPC } from '../rules/lambda';
+import {
+  OpenSearchEncryptedAtRest,
+  OpenSearchInVPCOnly,
+  OpenSearchNodeToNodeEncryption,
+} from '../rules/opensearch';
+import {
+  RDSInBackupPlan,
+  RDSInstanceDeletionProtectionEnabled,
+  RDSInstancePublicAccess,
+  RDSLoggingEnabled,
+  RDSMultiAZSupport,
+  RDSStorageEncrypted,
+} from '../rules/rds';
+import {
+  RedshiftRequireTlsSSL,
+  RedshiftBackupEnabled,
+  RedshiftClusterConfiguration,
+  RedshiftClusterMaintenanceSettings,
+  RedshiftClusterPublicAccess,
+} from '../rules/redshift';
+import {
+  S3BucketLevelPublicAccessProhibited,
+  S3BucketLoggingEnabled,
+  S3BucketPublicReadProhibited,
+  S3BucketPublicWriteProhibited,
+  S3BucketServerSideEncryptionEnabled,
+  S3BucketSSLRequestsOnly,
+  S3BucketVersioningEnabled,
+  S3DefaultEncryptionKMS,
+} from '../rules/s3';
+import {
+  SageMakerEndpointConfigurationKMSKeyConfigured,
+  SageMakerNotebookInstanceKMSKeyConfigured,
+  SageMakerNotebookNoDirectInternetAccess,
+} from '../rules/sagemaker';
+import { SecretsManagerUsingKMSKey } from '../rules/secretsmanager';
+import { SNSEncryptedKMS } from '../rules/sns';
+import {
+  VPCFlowLogsEnabled,
+  VPCDefaultSecurityGroupClosed,
+} from '../rules/vpc';
+import { WAFv2LoggingEnabled } from '../rules/waf';
+
+/**
+ * Check for NZISM v36-1022-20 compliance.
+ * Based on the NZISM v36-1022-20: <URL To be updated once published>
+ *
+ * The following rules are not yet included and are TODO items.
+ *
+ * CloudfrontDefaultRootObjectConfigured
+ * DynamodbTableEncryptedKms
+ * EbsSnapshotPublicRestorableCheck
+ * Ec2HardeningAndStateManagement
+ * Ec2Imdsv2Check
+ * EcsContainersReadonlyAccess
+ * ElbCustomSecurityPolicySslCheck
+ * EmrMasterNoPublicIp
+ * RdsSnapshotEncrypted
+ * VpcSgOpenOnlyToAuthorizedPorts
```

Comment:
Some of these rules can't be determined at the CDK level, we will have to investigate further. Some of these rules have been [excluded in other Packs](https://github.com/cdklabs/cdk-nag/blob/main/RULES.md#excluded-rules-2) because they couldn't be implemented

Reply:
A full list of the inclusions is now in the RULES.md..   I am wondering if there needs to be a couple of types of 'ignored' rules..    ones that are ignored because they are account level, and then some because a rule does not exisit yet.

Reply:
**For context:**

Ideally the excludes should only be rules that are ignored because 

1. They are account level
2. Can't be implemented due to CloudFormation not supporting it
3. Are too dependent on business logic (ex. "Don't use these specific arbitrary ports")

Ideally if there are rules that haven't been implemented, we'd resolve those by  actually implementing them before release. There also  may be rules that may be rules that were overlooked or can now be implemented due to more recent CloudFormation support.


**Actual Answer**

We could probably separate them out, but I think if we give any accurate description on the summary of the rule it should be enough context.